### PR TITLE
Bump Poetry to `1.2`

### PR DESCRIPTION
**Pull request by parisk**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,17 +25,17 @@ This will use the default Python version (latest) and image variant (Buster). Yo
...
+- Poetry 1.2 on Buster: `ghcr.io/withlogicco/poetry:1.2-buster`
+- Poetry 1.2 on Python 3.10 and Buster: `ghcr.io/withlogicco/poetry:1.2-python-3.10-buster`
 
 ## Usage
 
 First, pick the image tag you wish to use. Then, copy your `pyproject.toml` and `poetry.lock` files in the working directory (`/usr/src/app`), for optimal caching and run `poetry install`. Example:
 
 ```dockerfile
-FROM ghcr.io/withlogicco/poetry:1.1.14
+FROM ghcr.io/withlogicco/poetry:1.2
```
Comment (akalipetis) [inline README.md:38]:
```suggestion
FROM ghcr.io/withlogicco/poetry:1.2.0
```

File: README.md [hunk 2]
```diff
@@ -25,17 +25,17 @@ This will use the default Python version (latest) and image variant (Buster). Yo
...
 ### Examples
 
-- Poetry 1.1.14: `ghcr.io/withlogicco/poetry:1.1.14`
-- Poetry 1.1.14 on Python 3.10: `ghcr.io/withlogicco/poetry:1.1.14-python-3.10`
-- Poetry 1.1.14 on Buster: `ghcr.io/withlogicco/poetry:1.1.14-buster`
-- Poetry 1.1.14 on Python 3.10 and Buster: `ghcr.io/withlogicco/poetry:1.1.14-python-3.10-buster`
+- Poetry 1.2: `ghcr.io/withlogicco/poetry:1.2`
+- Poetry 1.2 on Python 3.10: `ghcr.io/withlogicco/poetry:1.2-python-3.10`
+- Poetry 1.2 on Buster: `ghcr.io/withlogicco/poetry:1.2-buster`
+- Poetry 1.2 on Python 3.10 and Buster: `ghcr.io/withlogicco/poetry:1.2-python-3.10-buster`
```
Comment (akalipetis) [inline README.md:31]:
```suggestion
- Poetry 1.2.0: `ghcr.io/withlogicco/poetry:1.2.0`
- Poetry 1.2.0 on Python 3.10: `ghcr.io/withlogicco/poetry:1.2.0-python-3.10`
- Poetry 1.2.0 on Buster: `ghcr.io/withlogicco/poetry:1.2.0-buster`
- Poetry 1.2.0 on Python 3.10 and Buster: `ghcr.io/withlogicco/poetry:1.2.0-python-3.10-buster`
```

File: README.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 Docker Poetry is a Docker image for Python projects with dependencies managed by Poetry.
 
-![Poetry](https://img.shields.io/badge/Poetry-1.1.14-purple) ![Python](https://img.shields.io/badge/Python-3.10%20(default)%20%7C%203.9%20%7C%203.8%20%7C%203.7-blue) ![Variants](https://img.shields.io/badge/Variants-Buster%20(default)%20%7C%20Slim%20%7C%20Alpine-orange
+![Poetry](https://img.shields.io/badge/Poetry-1.2-purple) ![Python](https://img.shields.io/badge/Python-3.10%20(default)%20%7C%203.9%20%7C%203.8%20%7C%203.7-blue) ![Variants](https://img.shields.io/badge/Variants-Buster%20(default)%20%7C%20Slim%20%7C%20Alpine-orange
```
Comment (akalipetis) [inline README.md:5]:
```suggestion
![Poetry](https://img.shields.io/badge/Poetry-1.2.0-purple) ![Python](https://img.shields.io/badge/Python-3.10%20(default)%20%7C%203.9%20%7C%203.8%20%7C%203.7-blue) ![Variants](https://img.shields.io/badge/Variants-Buster%20(default)%20%7C%20Slim%20%7C%20Alpine-orange
```

**Review by akalipetis — CHANGES_REQUESTED** [review]
Do we have a reason not to pin the patch version, like we previously did?

Also, let's prefer the official installer.

File: Dockerfile [hunk 4]
```diff
@@ -1,27 +1,13 @@
...
-RUN pip install --upgrade pip==${PIP_VERSION}
+ARG SETUPTOOLS_VERSION=65.3.0
+RUN pip install --upgrade pip==${PIP_VERSION} setuptools==${SETUPTOOLS_VERSION}
 
-ARG POETRY_HOME=/usr/local/poetry
-ENV POETRY_HOME=${POETRY_HOME}
-COPY --from=installation-base ${POETRY_HOME} ${POETRY_HOME}
-RUN ln -s ${POETRY_HOME}/bin/poetry /usr/local/bin/poetry
+ARG POETRY_VERSION=1.2
+RUN pip install poetry==${POETRY_VERSION}
```
[the remark this answers is not in the document]
Comment (parisk) [reply Dockerfile:13]: Installing with `pip` is officially supported and looks like a better option, as described in both the manual[^1] and the CI[^2], where it states

> Unlike development environments, where making use of the latest tools is desirable, in a CI environment reproducibility should be made the priority.

[^1]: https://python-poetry.org/docs/#installing-manually
[^2]: https://python-poetry.org/docs/#ci-recommendations

**Pull request merged**